### DISCUSSION TITLE
Printing ASCII Characters in a Sprite with a ordinal number less than 32

### DISCUSSION THREAD

**Discussion by DGhost001** [Bodmer/TFT_eSPI#3042] [Q&A - Fonts]
Hi,

I use the library with a full 256 Character font. So there are natural also the character symbols with ordinal numbers between 0..31 available.

If I want to print a text that contain any of those characters, they do not show up when the text is printed into a sprite.

The issue is in Sprite.cpp [here](https://github.com/Bodmer/TFT_eSPI/blob/fa6512faf40594fde13795074e26ef0379708f38/Extensions/Sprite.cpp#L1988C22-L1988C22)
If I see this correctly the `if (c<32) return;` is not needed and prevents the usage of characters less than 32.

So I would suggest to remove it, or check the availability of the characters in the current font here.

Cheers

**Comment by Bodmer** [comment]
ASCII characters less that 32 are control codes and thus are not printable characters and do not exist in fonts.

**Reply by DGhost001** [reply]
I'm afraid this is not completely true. The symbols below 32 have a control code meaning attached to them yes, but they also have symbols attached to them.

See for example the [Codpage 850](https://en.wikipedia.org/wiki/Code_page_850#/media/File:Codepage-850.png) on wikipedia.

There are those nice smilies on character 0x1 and 0x2, but there are also those nice arrows or game card symbols.

A font with all characters populated also those below 32, you can find for example [here](https://github.com/DGhost001/passkey/blob/master/src/ec1834.cpp) in my github repo if you are interested.

The font I use is a rom dump of a real font ROM from a PC and it has those symbols, and I would like to make use of them. Especially the arrow symbols.

**Reply by Bodmer** [reply]
ASCII only has 95 printable characters, [see here](https://en.wikipedia.org/wiki/ASCII). There are other historical encoding schemes for glyphs and codes, but these are not supported. The aim is to support the Unicode standard, which adopts ASCII.

**Reply by DGhost001** [reply]
Unicode would be also fine for me. But the Symbols: ►, ◄, ↑, ↓,→,←, ▲, ▼ are note mapped correctly.
The unicode specified alt codes for those symbols are 0x10, 0x11, 0x18, 0x19, 0x1A, 0x1B, 0x1E and 0x1F ....
 So it would be great if they are mapped in the "unused" character space below 32 at the alt code places defined by the unicode standard.

**Reply by Bodmer** [reply]
The Alt codes are mapped to higher code point values in Unicode. You can see the [mappings here](https://en.wikipedia.org/wiki/Code_page_437) for the CP437 character set. So ▲ is U+25B2 and is printable by this library if the font includes it.

It is not appropriate for the library to support different legacy encodings for control codes, however you should find all the symbols you require are in the Unicode glyph set.
